extensions/wikia/AdEngine/js/SourcePointDetection.js

```diff
@@ -62,20 +63,22 @@ define('ext.wikia.adEngine.sourcePointDetection', [
 		detectionScript.src = context.opts.sourcePointDetectionUrl;
 		detectionScript.setAttribute('data-client-id', getClientId());
 
-		doc.addEventListener('sp.blocking', function () {
-			window.ads.runtime.sp.blocking = true;
-			trackStatusOnce('yes');
-		});
-		doc.addEventListener('sp.not_blocking', function () {
-			window.ads.runtime.sp.blocking = false;
-			trackStatusOnce('no');
-		});
-
 		log('Appending detection script to head', 'debug', logGroup);
 		node.parentNode.insertBefore(detectionScript, node);
 		detectionInitialized = true;
 	}
 
+	win.ads.runtime.sp = win.ads.runtime.sp || {};
+
+	doc.addEventListener('sp.blocking', function () {
+		win.ads.runtime.sp.blocking = true;
+		trackStatusOnce('yes');
+	});
+	doc.addEventListener('sp.not_blocking', function () {
+		win.ads.runtime.sp.blocking = false;
+		trackStatusOnce('no');
+	});
```

Comment:
I don't like the thing that the code is executed just during module loading.
What do you think about wrapping this code in the method and call the method outside of this module?